File: app/src/main/java/com/goran/quranipiroz/utils/tafsir/TafsirJSInterface.java
package com.goran.quranipiroz.utils.tafsir;

import android.content.ComponentName;
import android.content.Intent;
import android.webkit.JavascriptInterface;

import com.goran.quranipiroz.activities.ActivityReader;
import com.goran.quranipiroz.activities.ActivityTafsir2;
import com.goran.quranipiroz.utils.reader.factory.ReaderFactory;
import com.goran.quranipiroz.utils.univ.Codes;
import com.goran.quranipiroz.utils.univ.Keys;

public class TafsirJSInterface {
    private final ActivityTafsir2 mActivityTafsir;

    public TafsirJSInterface(ActivityTafsir2 activityTafsir) {
        mActivityTafsir = activityTafsir;
    }

    @JavascriptInterface
    public void goToVerse() {
        Intent intent = ReaderFactory.prepareSingleVerseIntent(mActivityTafsir.mChapterNo, mActivityTafsir.mVerseNo);

        ComponentName callingActivity = mActivityTafsir.getCallingActivity();
        if (callingActivity != null && ActivityReader.class.getName().equals(callingActivity.getClassName())) {
            mActivityTafsir.setResult(Codes.OPEN_REFERENCE_RESULT_CODE, intent);
            mActivityTafsir.finish();
        } else {
            intent = intent.setClass(mActivityTafsir, ActivityReader.class);
            mActivityTafsir.startActivity(intent);
        }
    }

    @JavascriptInterface
    public void previousTafsir() {
        if (mActivityTafsir.mVerseNo == 1) {
            return;
        }
        Intent intent = mActivityTafsir.getIntent();
        intent.putExtra(Keys.READER_KEY_VERSE_NO, mActivityTafsir.mVerseNo - 1);
        intent.setAction(null);
        mActivityTafsir.startActivity(intent);
    }

    @JavascriptInterface
    public void nextTafsir() {
        if (mActivityTafsir.mVerseNo == mActivityTafsir.mQuranMeta.getChapterVerseCount(mActivityTafsir.mChapterNo)) {
            return;
        }

        Intent intent = mActivityTafsir.getIntent();
        intent.putExtra(Keys.READER_KEY_VERSE_NO, mActivityTafsir.mVerseNo + 1);
        intent.setAction(null);
        mActivityTafsir.startActivity(intent);
    }
}

File: app/src/main/java/com/goran/quran/karim/dang/utils/tafsir/TafsirJsInterface.kt
package com.goran.quran.karim.dang.utils.tafsir

import android.webkit.JavascriptInterface
import com.goran.quran.karim.dang.activities.ActivityReader
import com.goran.quran.karim.dang.activities.ActivityTafsir
import com.goran.quran.karim.dang.utils.reader.factory.ReaderFactory.prepareSingleVerseIntent
import com.goran.quran.karim.dang.utils.univ.Codes
import com.goran.quran.karim.dang.utils.univ.Keys

class TafsirJsInterface(private val activity: ActivityTafsir) {
    @JavascriptInterface
    fun goToVerse() {
        var intent = prepareSingleVerseIntent(activity.chapterNo, activity.verseNo)

        val callingActivity = activity.callingActivity
        if (callingActivity != null && ActivityReader::class.java.name == callingActivity.className) {
            activity.setResult(Codes.OPEN_REFERENCE_RESULT_CODE, intent)
            activity.finish()
        } else {
            intent = intent.setClass(activity, ActivityReader::class.java)
            activity.startActivity(intent)
        }
    }


    @JavascriptInterface
    fun previousTafsir() {
        if (activity.verseNo == 1) {
            return
        }

        val intent = activity.intent.also {
            it.putExtra(Keys.READER_KEY_VERSE_NO, activity.verseNo - 1)
            it.action = null
        }

        activity.startActivity(intent)
    }

    @JavascriptInterface
    fun nextTafsir() {
        if (activity.verseNo == activity.mQuranMetaRef.get()!!.getChapterVerseCount(activity.chapterNo)) {
            return
        }

        val intent = activity.intent.also {
            it.putExtra(Keys.READER_KEY_VERSE_NO, activity.verseNo + 1)
            it.action = null
        }

        activity.startActivity(intent)
    }

    @JavascriptInterface
    fun goToTop() {
        activity.scrollToTop()
    }
}